pydocstringformatter/run.py

```diff
@@ -123,11 +113,117 @@ def get_enabled_formatters(self) -> dict[str, _formatting.Formatter]:
+                new_tokeninfo, changers, stable = self.apply_formatters_twice(
+                    new_tokeninfo,
+                )
+
+                if not stable:
+                    if len(changers) > 1:
+                        msg = "Conflicting formatters: " + ", ".join(changers)
+                    else:
+                        msg = "Formatter: " + changers[0]
+                        msg += "Is not able to make stable changes on:\n"
+                        msg += new_tokeninfo.string + "\n"
+
+                    if len(changers) == 2:
+                        diff = compare_formatters(
+                            new_tokeninfo,
+                            self.enabled_formatters[changers[0]],
+                            self.enabled_formatters[changers[1]],
+                            str(filename),
+                        )
+
+                        msg += f"\n{diff}"
+
+                    raise RuntimeError(msg)
+
+            formatted_tokens.append(new_tokeninfo)
+
+        return formatted_tokens, is_changed
+
+    def apply_formatters_twice(
+        self, token: tokenize.TokenInfo, allowed_formatters: list[str] | None = None
+    ) -> tuple[tokenize.TokenInfo, list[str], bool]:
+        """Apply the formatters twice to a token.
+
+        Also tracks which formatters changed the token.
+
+        Returns:
+            A tuple containing:
+            [1] the formatted token and
+            [2] a list of formatters that changed the token.
+            [3] a boolean indicating if the token was in the second pass.
+        """
+        stable = False
+        token, changers = self.apply_formatters(token, allowed_formatters)
+        if changers:
+            token, changers2 = self.apply_formatters(token, allowed_formatters)
+            stable = not bool(changers2)
+            changers = list(set(changers + changers2))
+        else:
+            stable = True
+        return token, changers, stable
+
+    def apply_formatters(
+        self, token: tokenize.TokenInfo, allowed_formatters: list[str] | None = None
+    ) -> tuple[tokenize.TokenInfo, list[str]]:
+        """Applies formatters to a token and keeps track of what changes it.
+
+        token: Token to apply formatters to
+        allowed_formatters:
+            List of formatters to apply.
+            If None, all enabled formatters are applied.
+
+        Returns:
+            A tuple containing [1] the formatted token and [2] a list
+            of formatters that changed the token.
+        """
+        changers = []
+        for formatter_name, formatter in self.enabled_formatters.items():
+            if allowed_formatters and formatter_name not in allowed_formatters:
+                continue
+
+            if (new_token := formatter.treat_token(token)) != token:
+                changers.append(formatter_name)
+                token = new_token
+
+        return token, changers
+
+    def format_files(self, filepaths: list[Path]) -> bool:
```

Comment:
Just to get the diff stable we might also want to do this refactor in a separate PR.